### PR TITLE
[BYOC][Verilator] add support to dynamically load hardware library

### PR DESCRIPTION
**Pull request by vegaluisjose**
@liangfu @tmoreau89

### CODE REVIEW

**Comment by tmoreau89** [comment]
@tqchen I'd love some input on this if you have time to review

**Review by liangfu — APPROVED** [review]

File: tests/python/contrib/test_verilator/infrastructure.py [hunk 1]
```diff
@@ -66,10 +68,43 @@ def offload(mod):
     return mod
 
 
+def verilator_app_path():
```
Comment (liangfu) [inline tests/python/contrib/test_verilator/infrastructure.py:71]: I don't think it's actually running in the CI (see https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-7286/6/pipeline/255#step-340-log-91), would be better if we could enable testing this.

Comment (vegaluisjose) [reply tests/python/contrib/test_verilator/infrastructure.py:71]: Hey @liangfu

Yeah that is the plan, once we merge this. The idea is to finally update that PR that we have open for CI.

**Review by tmoreau89 — APPROVED** [review]
Love the new dynamic load feature, the code below looks very clean.
```
with tvm.transform.PassContext(
        opt_level=3, config={"relay.ext.verilator.options": {"lib": lib}}
    ):
```
LGTM

**Pull request merged**